Build failure: pkgsCross.m68k.newlib

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```console
$ nix-build '<nixpkgs>' -A pkgsCross.m68k.newlib
```

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
[…]
make[4]: Entering directory '/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k'
m68k-unknown-linux-gnu-cc -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/ -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/targ-include -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/newlib/libc/include -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/libnosys -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/libgloss/m68k    -g -O2 -I. -I../../.././libgloss/m68k/.. -I./.. -idirafter ../../.././libgloss/m68k/../../include -g -O2 -c ../../.././libgloss/m68k/sim-crt0.S
m68k-unknown-linux-gnu-cc -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/ -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/targ-include -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/newlib/libc/include -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/libnosys -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/libgloss/m68k    -g -O2 -O2 -I. -I../../.././libgloss/m68k/.. -I./.. -idirafter ../../.././libgloss/m68k/../../include -c -g -O2 ../../.././libgloss/m68k/sim-print.c
m68k-unknown-linux-gnu-cc -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/ -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/targ-include -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/newlib/libc/include -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/libnosys -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/libgloss/m68k    -g -O2 -O2 -I. -I../../.././libgloss/m68k/.. -I./.. -idirafter ../../.././libgloss/m68k/../../include -c -g -O2 ../../.././libgloss/m68k/sim-inbyte.c
m68k-unknown-linux-gnu-cc -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/ -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/targ-include -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/newlib/libc/include -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/libnosys -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/libgloss/m68k    -g -O2 -O2 -I. -I../../.././libgloss/m68k/.. -I./.. -idirafter ../../.././libgloss/m68k/../../include -c -g -O2 ../../.././libgloss/m68k/sim-sbrk.c
m68k-unknown-linux-gnu-cc -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/ -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/targ-include -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/newlib/libc/include -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/libnosys -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/libgloss/m68k    -g -O2 -O2 -I. -I../../.././libgloss/m68k/.. -I./.. -idirafter ../../.././libgloss/m68k/../../include -c -g -O2 ../../.././libgloss/m68k/sim-abort.c
m68k-unknown-linux-gnu-cc -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/ -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/newlib/targ-include -isystem /nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/newlib/libc/include -B/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/libnosys -L/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/libgloss/m68k    -g -O2 -O2 -I. -I../../.././libgloss/m68k/.. -I./.. -idirafter ../../.././libgloss/m68k/../../include -c -g -O2 ../../.././libgloss/m68k/sim-errno.c
../../.././libgloss/m68k/sim-print.c: In function 'print':
../../.././libgloss/m68k/sim-print.c:20:1: warning: old-style function definition [-Wold-style-definition]
   20 | print (ptr)
      | ^~~~~
../../.././libgloss/m68k/sim-sbrk.c: In function 'sbrk':
../../.././libgloss/m68k/sim-sbrk.c:21:1: warning: old-style function definition [-Wold-style-definition]
   21 | sbrk (incr)
      | ^~~~
../../.././libgloss/m68k/sim-sbrk.c:26:7: error: implicit declaration of function 'brk'; did you mean 'sbrk'? [-Wimplicit-function-declaration]
   26 |   if (brk (curbrk) == -1)
      |       ^~~
      |       sbrk
../../.././libgloss/m68k/sim-print.c:29:3: error: implicit declaration of function 'write' [-Wimplicit-function-declaration]
   29 |   write (1, ptr, p-ptr);
      |   ^~~~~
../../.././libgloss/m68k/sim-print.c: In function 'outbyte':
../../.././libgloss/m68k/sim-print.c:37:1: warning: old-style function definition [-Wold-style-definition]
   37 | outbyte (c_int)
      | ^~~~~~~
../../.././libgloss/m68k/sim-inbyte.c: In function 'inbyte':
../../.././libgloss/m68k/sim-inbyte.c:22:7: error: implicit declaration of function 'read' [-Wimplicit-function-declaration]
   22 |   if (read (0, &c, 1) <= 0)
      |       ^~~~
make[4]: *** [Makefile:212: sim-inbyte.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: *** [Makefile:212: sim-sbrk.o] Error 1
make[4]: *** [Makefile:212: sim-print.o] Error 1
../../.././libgloss/m68k/sim-abort.c: In function 'abort':
../../.././libgloss/m68k/sim-abort.c:19:3: error: implicit declaration of function 'write' [-Wimplicit-function-declaration]
   19 |   write (2, "Abort called.\n", sizeof("Abort called.\n")-1);
      |   ^~~~~
../../.././libgloss/m68k/sim-abort.c:20:3: error: implicit declaration of function 'exit' [-Wimplicit-function-declaration]
   20 |   exit (1);
      |   ^~~~
../../.././libgloss/m68k/sim-abort.c:1:1: note: include '<stdlib.h>' or provide a declaration of 'exit'
  +++ |+#include <stdlib.h>
    1 | /*
../../.././libgloss/m68k/sim-abort.c:20:3: warning: incompatible implicit declaration of built-in function 'exit' [-Wbuiltin-declaration-mismatch]
   20 |   exit (1);
      |   ^~~~
../../.././libgloss/m68k/sim-abort.c:20:3: note: include '<stdlib.h>' or provide a declaration of 'exit'
make[4]: *** [Makefile:212: sim-abort.o] Error 1
../../.././libgloss/m68k/sim-errno.c: In function '_cerror':
../../.././libgloss/m68k/sim-errno.c:23:1: warning: old-style function definition [-Wold-style-definition]
   23 | _cerror (e)
      | ^~~~~~~
make[4]: Leaving directory '/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss/m68k'
make[3]: *** [Makefile:9835: all-recursive] Error 1
make[3]: Leaving directory '/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss'
make[2]: *** [Makefile:3158: all] Error 2
make[2]: Leaving directory '/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231/m68k-unknown-linux-gnu/libgloss'
make[1]: *** [Makefile:9529: all-target-libgloss] Error 2
make[1]: Leaving directory '/nix/var/nix/builds/nix-63849-1370409654/newlib-4.5.0.20241231'
make: *** [Makefile:880: all] Error 2
```

### Additional context

I was actually trying to build for `m68k-none-elf` when I first discovered this issue:
```console
$ nix-build '<nixpkgs>' --arg crossSystem '{ config="m68k-none-elf"; }' -A newlib
[…fails in the same way…]
```

Seems to have been [fixed upstream](https://www.sourceware.org/git/?p=newlib-cygwin.git;h=22580ced9ac9af2a1e3068863bebecf738e33458;hp=d61692cbd03baf863b91d23bb3816ce2e891dcc2), but there hasn't been a new source snapshot since then. Adding that commit as [a patch](https://www.sourceware.org/git/?p=newlib-cygwin.git;a=patch;h=22580ced9ac9af2a1e3068863bebecf738e33458;hp=d61692cbd03baf863b91d23bb3816ce2e891dcc2) ([like so](https://github.com/Rhys-T/nur-packages/blob/6239cb18ebc524cf93b06c86ce02e60fead027ef/pkgs/pce/generic.nix#L67-L71)) seems to make it build properly.

### System metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/q7v9hjls01b066fb28df3jpxbz7zxfm3-nixpkgs/nixpkgs`

### Notify maintainers

No maintainers listed for `newlib`. Most recent `newlib`-specific change seems to have been by @OPNA2608, fixing the MIPS equivalent of this issue (#406428, #407132).

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ne/newlib/package.nix
+++ b/pkgs/by-name/ne/newlib/package.nix
@@ -13,26 +13,14 @@
 
 stdenvNoLibc.mkDerivation (finalAttrs: {
   pname = "newlib";
-  version = "4.5.0.20241231";
+  version = "4.6.0.20260123";
 
   src = fetchurl {
     url = "ftp://sourceware.org/pub/newlib/newlib-${finalAttrs.version}.tar.gz";
-    sha256 = "sha256-M/EmBeAFSWWZbCXBOCs+RjsK+ReZAB9buMBjDy7IyFI=";
+    sha256 = "sha256-b/J+O/AiZm9D94AiVb5oDu/3IqwYGxcl0h4ugxhgTuM=";
   };
 
-  patches = [
-    (fetchpatch {
-      name = "0001-newlib-Fix-mips-libgloss-support.patch";
-      url = "https://sourceware.org/git/?p=newlib-cygwin.git;a=patch;h=8a8fb570d7c5310a03a34b3dd6f9f8bb35ee9f40";
-      hash = "sha256-hWS/X0jf/ZFXIR39NvNDVhkR8F81k9UWpsqDhZFxO5o=";
-    })
-    (fetchpatch {
-      name = "0002-newlib-Fix-i386-libgloss-support.patch";
-      url = "https://sourceware.org/git/?p=newlib-cygwin.git;a=patch;h=351842d55ee50fab672818b72f2c6bf3be1772bd;hp=4a1144dc540423b422dbec27eb433bc7dc725ae8";
-      hash = "sha256-GMx9nQKJEnXbA5HFczp3xjrrU94j1oXq4lIjEf0um9Y=";
-    })
-  ]
-  ++ lib.optionals nanoizeNewlib [
+  patches = lib.optionals nanoizeNewlib [
     # https://bugs.gentoo.org/723756
     (fetchpatch {
       name = "newlib-3.3.0-no-nano-cxx.patch";
@@ -114,6 +102,8 @@ stdenvNoLibc.mkDerivation (finalAttrs: {
   );
 
   enableParallelBuilding = true;
+  # install: cannot create regular file '.../powerpc-none-eabi/lib/crt0.o': File exists
+  enableParallelInstalling = false;
   dontDisableStatic = true;
 
   # apply necessary nano changes from https://developer.arm.com/-/media/Files/downloads/gnu/12.2.rel1/manifest/copy_nano_libraries.sh?rev=4c50be6ccb9c4205a5262a3925317073&hash=1375A7B0A1CD0DB9B9EB0D2B574ADF66

